components/collector/tests/source_collectors/cargo_audit/test_security_warnings.py

```diff
@@ -0,0 +1,48 @@
+"""Unit tests for the Cargo Audit security warnings collector."""
+
+from tests.source_collectors.source_collector_test_case import SourceCollectorTestCase
+
+
+class CargoAuditSecurityWarningsTest(SourceCollectorTestCase):
+    """Unit tests for the security warning metric."""
+
+    SOURCE_TYPE = "cargo_audit"
+    METRIC_TYPE = "security_warnings"
+
+    async def test_warnings(self):
+        """Test the number of security warnings."""
+        vulnerabilities_json = {
+            "vulnerabilities": {
+                "list": [
+                    {
+                        "advisory": {
+                            "id": "RUSTSEC-2022-0090",
+                            "title": "`libsqlite3-sys` via C SQLite CVE-2022-35737",
```

Comment:
Should we remove special chars, or at least those backticks from the title? It's interesting that this title does not exist anywhere on the advisory url btw, is it composed by cargo audit?

Reply:
I noted one of the descriptions had Markdown too, but the description of another vulnerability didn't. I don't know where the Markdown comes from. But, as it is Markdown, it's ok to keep it in I think.